src/components/button/button.spec.js

```diff
@@ -0,0 +1,325 @@
+      const wrapper = TestRenderer.create(<StyledButton disabled theme={ classicTheme }>foo</StyledButton>);
+      assertStyleMatch({
+        background: '#e6ebed'
+      }, wrapper.toJSON());
+    });
+  });
+
+  describe('A primary button', () => {
+    const primary = render({
+      name: 'Primary Button',
+      as: 'primary',
+      onClick: jest.fn(),
+      children: 'Primary'
+    }).dive();
+
+    it('renders a primary button', () => {
+      expect(primary.props().name).toEqual('Primary Button');
+      expect(primary.props().buttonType).toEqual('primary');
+      expect(primary.containsMatchingElement(
+        <span>Primary</span>
+      )).toBeTruthy();
+    });
+  });
+
+  describe('A secondary button', () => {
+    const secondary = render({
+      name: 'Secondary Button',
+      className: 'customClass',
+      theme: 'red',
+      children: 'Secondary'
+    }).dive();
+
+    it('renders a secondary button', () => {
+      expect(secondary.props().name).toEqual('Secondary Button');
+      expect(secondary.props().buttonType).toEqual('secondary');
+      expect(secondary.containsMatchingElement(
+        <span>Secondary</span>
+      )).toBeTruthy();
+    });
+  });
+
+  describe('A small button', () => {
+    const small = render({
+      name: 'Small Button',
+      size: 'small',
+      children: 'Small'
+    }).dive();
+
+    it('renders a small button', () => {
+      expect(small.props().name).toEqual('Small Button');
+      expect(small.props().size).toEqual('small');
+      expect(small.containsMatchingElement(
+        <span>Small</span>
+      )).toBeTruthy();
+    });
+  });
+
+  describe('A large button', () => {
+    const large = render({
+      name: 'Large Button',
+      size: 'large',
+      children: 'Large'
+    }).dive();
+
+    it('renders a large button', () => {
+      expect(large.props().name).toEqual('Large Button');
+      expect(large.props().size).toEqual('large');
+      expect(large.containsMatchingElement(
+        <span>Large</span>
+      )).toBeTruthy();
+    });
+  });
+
+  describe('A disabled button', () => {
+    const disabled = render({
+      name: 'Disabled Button',
+      disabled: true,
+      children: 'Disabled'
+    }).dive();
+
+    it('renders a disabled button', () => {
+      expect(disabled.props().name).toEqual('Disabled Button');
+      expect(disabled.props().buttonType).toEqual('secondary');
+      expect(disabled.containsMatchingElement(
+        <span>Disabled</span>
+      )).toBeTruthy();
+      expect(disabled.props().disabled).toEqual(true);
+    });
+  });
+
+  describe('when a subtext prop is passed into the component', () => {
+    it('does not render the subtext if the size prop is not "large"', () => {
+      try {
+        const wrapper = render({ children: 'foo', subtext: 'bar' }).dive();
+
+        expect(wrapper.find('[data-element="subtext"]').exists()).toBe(false);
+      } catch (error) {} // eslint-disable-line no-empty
```

Comment:
Is it worth logging the error?

Reply:
this code just catches that error so it won't be logged in the console (which we do not want).